File: src/main/python/leetcode/_1_E.py
from typing import List, Dict


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        existed: Dict[int, int] = {}
        for index, num in enumerate(nums):
            existedIndex = existed.get(target - num)
            if existedIndex is not None:
                return [existedIndex, index]
            existed[num] = index
        return [-1, -1]

File: typescript/src/leetcode/_1_E.ts
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
function twoSum(nums: number[], target: number): [number, number] {
    const existed = {};
    for (let i = 0, len = nums.length; i < len; i++) {
        const num = nums[i];
        const existedIndex = existed[target - num];
        if (existedIndex != null) {
            return [existedIndex, i];
        }
        existed[num] = i;
    }
    return [-1, -1];
}
